bundles/org.openhab.core/src/test/java/org/openhab/core/library/types/HSBTypeTest.java

```diff
@@ -198,4 +206,109 @@ public void testConstructorWithIllegalSaturationValue() {
+    public void parameterizedTestRgbToHsbToRgbConversion(int[] rgb, int maxSquaredSum) {
+        HSBType hsb = HSBType.fromRGB(rgb[0], rgb[1], rgb[2]);
+        Assertions.assertNotNull(hsb);
+
+        final PercentType[] rgbPercent = hsb.toRGB();
+        compareRgbToRgbValues(rgb, rgbPercent, maxSquaredSum);
+    }
+
+    // return a stream of well known HSB - RGB pairs
+    static class HsbRgbProvider implements ArgumentsProvider {
+        @Override
+        public Stream<? extends Arguments> provideArguments(@Nullable ExtensionContext context) throws Exception {
+            return Stream.of(Arguments.of(new int[] { 0, 0, 0 }, new int[] { 0, 0, 0 }),
+                    Arguments.of(new int[] { 0, 0, 100 }, new int[] { 255, 255, 255 }),
+                    Arguments.of(new int[] { 0, 100, 100 }, new int[] { 255, 0, 0 }),
+                    Arguments.of(new int[] { 120, 100, 100 }, new int[] { 0, 255, 0 }),
+                    Arguments.of(new int[] { 240, 100, 100 }, new int[] { 0, 0, 255 }),
+                    Arguments.of(new int[] { 60, 100, 100 }, new int[] { 255, 255, 0 }),
+                    Arguments.of(new int[] { 180, 100, 100 }, new int[] { 0, 255, 255 }),
+                    Arguments.of(new int[] { 300, 100, 100 }, new int[] { 255, 0, 255 }),
+                    Arguments.of(new int[] { 0, 0, 75 }, new int[] { 191, 191, 191 }),
+                    Arguments.of(new int[] { 0, 0, 50 }, new int[] { 128, 128, 128 }),
+                    Arguments.of(new int[] { 0, 100, 50 }, new int[] { 128, 0, 0 }),
+                    Arguments.of(new int[] { 60, 100, 50 }, new int[] { 128, 128, 0 }),
+                    Arguments.of(new int[] { 120, 100, 50 }, new int[] { 0, 128, 0 }),
+                    Arguments.of(new int[] { 300, 100, 50 }, new int[] { 128, 0, 128 }),
+                    Arguments.of(new int[] { 180, 100, 50 }, new int[] { 0, 128, 128 }),
+                    Arguments.of(new int[] { 240, 100, 50 }, new int[] { 0, 0, 128 }));
+        }
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(HsbRgbProvider.class)
+    public void paremeterizedTestHsbToRgbConversion(int[] hsb, int[] rgb) {
+        final String hsbString = hsb[0] + ", " + hsb[1] + ", " + hsb[2];
+        final HSBType hsbType = new HSBType(hsbString);
+
+        final PercentType[] rgbPercent = hsbType.toRGB();
+        compareRgbToRgbValues(rgb, rgbPercent, 0);
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(HsbRgbProvider.class)
+    public void paremeterizedTestRgbToRgbConversion(int[] hsb, int[] rgb) {
+        final HSBType hsbType = HSBType.fromRGB(rgb[0], rgb[1], rgb[2]);
+
+        final PercentType[] rgbPercent = hsbType.toRGB();
+        compareRgbToRgbValues(rgb, rgbPercent, 0);
+    }
+
+    // compare two rgb representations based on squared sum of differences, prepare a readable test message on fail
+    private void compareRgbToRgbValues(final int[] rgb, final PercentType[] rgbPercent, int maxSquaredSum) {
```

Comment:
```suggestion
    private void assertEqualsRGB(final int[] rgb, final PercentType[] rgbPercent, int maxSquaredSum) {
```

Reply:
Thanks @J-N-K , I will have a look tomorrow. 

Should I move the whole RGB/HSV conversion to ```ColorUtil```` together with the suggested function above?

Maybe it would be useful to deprecate the getter functions for single color channels like ```HSBType::getRed()```. The implementation is highly inefficient (converting to RGB and picking just one channel), especially when the users of this function calls the getters for all channels separately. getRed() is used ~50x in bindings (w/o test).